File: FMS5776_2075_3798/FMS5776_2075_3798/Student.cpp
#include "Student.h"

Student::Student(string & id, string & _firstName, string & _lastName, int _grade, string & _street, int _houseNo, string & _city, string & _zipCode, string & _phoneNo)
{
	strncpy_s(this->ID, 10, id.c_str(), 9);
	strncpy_s(this->firstName, 11, _firstName.c_str(), 10);
	strncpy_s(this->lastName, 10, _lastName.c_str(), 9);
	this->grade = _grade;
	strncpy_s(this->street, 12, _street.c_str(), 11);
	this->houseNo = _houseNo;
	strncpy_s(this->city, 12, _city.c_str(), 11);
	strncpy_s(this->zipCode, 8, _zipCode.c_str(), 7);
	strncpy_s(this->phoneNo, 11, _phoneNo.c_str(), 10);
}

ostream & operator<<(ostream & out, Student & stu)
{
	out << stu.ID << "\t" << stu.firstName << "\t" << stu.lastName << "\t" << stu.grade << "\t" << stu.street << "\t" << stu.houseNo << "\t" << stu.city << "\t" << stu.zipCode << "\t" << stu.phoneNo << endl;
	return out;
}

File: FMS5776_2075_3798/FMS_adapter/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace FMS_adapter
{
    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
    public class Student
    {
        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 10)]
        string id;
        public string ID { get { return id; } set { id = value; } }

        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 11)]
        string firstName;
        public string FirstName { get { return firstName; } set { firstName = value; } }

        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 12)]
        string lastName;
        public string LastName { get { return lastName; } set { lastName = value; } }

        int grade;
        public int Grade { get { return grade; } set { grade = value; } }

        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 12)]
        string street;
        public string Street { get { return street; } set { street = value; } }

        int houseNo;
        public int HouseNo { get { return houseNo; } set { houseNo = value; } }

        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 12)]
        string city;
        public string City { get { return city; } set { city = value; } }

        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 8)]
        string zipCode;
        public string ZipCode { get { return zipCode; } set { zipCode = value; } }

        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 11)]
        string phoneNo;
        public string PhoneNo { get { return phoneNo; } set { phoneNo = value; } }

        IntPtr myStudentPtr;
        public Student()
        {
            myStudentPtr = cppToCsharpAdapter.makeStudentObject();
        }
        ~Student()
        {
            //if (myStudentPtr != null)
            //    cppToCsharpAdapter.deleteStudentObject(ref myStudentPtr);
        }
    }
}
